src/prefect/deployments/deployments.py

```diff
@@ -213,10 +215,20 @@ async def load_flow_from_flow_run(
             storage_block = Block._from_block_document(storage_document)
         else:
             basepath = deployment.path or Path(deployment.manifest_path).parent
+            if runner_storage_base_path:
+                basepath = str(basepath).replace(
+                    "$STORAGE_BASE_PATH", runner_storage_base_path
+                )
+                logger.info(f"Using storage base path {basepath!r}")
             storage_block = LocalFileSystem(basepath=basepath)
 
-        logger.info(f"Downloading flow code from storage at {deployment.path!r}")
-        await storage_block.get_directory(from_path=deployment.path, local_path=".")
+        from_path = (
+            str(deployment.path).replace("$STORAGE_BASE_PATH", runner_storage_base_path)
+            if runner_storage_base_path and deployment.path
+            else deployment.path
+        )
+        logger.info(f"Downloading flow code from storage at {from_path!r}")
+        await storage_block.get_directory(from_path=from_path, local_path=".")
```

Comment:
This almost looks like we could make runners off-the-shelf compatible with agent-style storage, does that sound accurate to you?

Reply:
Runners should be able to execute flow runs with either a storage block or pull steps associated with their deployment since they follow the already established path. I'd rather have a utility that converts a storage block to runner storage than directly associate storage blocks with runner deployments.